### PR TITLE
Add an optional description field.

### PR DESCRIPTION
**Pull request by Wmaarts**
# Description field which contents are displayed in a modal dialog

### Why? 🤔
Right now to improve better pitest integration, but the description field can be used for other things in the future. 🌌

### Functional changes: ⚙
* Description is made an optional field for each mutator.
* Replacement is made optional and the replacement text will be filled
with the Mutatorname if it's not present.
* When a description is set a 'show more' button will appear that opens
a dialog which shows the description.
* Added emoji's to the mutantresult states. 🎉

### Technical changes: 🔧
* Pitest-example added based on results from [the pitest plugin](https://github.com/Wmaarts/pitest-mutation-testing-elements-plugin)

### Todo: 📋

- [ ] Integration / UI tests
- [ ] Update json schema + unit tests

### Preview: 👁
image

Clicking "Show more" opens this dialog.

image

### CODE REVIEW

**Comment by nicojs** [comment]
@hcoles what do you think about this UI? The `"description"` field was too large in some cases to show it in de popup directly. That's why we decided to present it in a modal dialog.

File: packages/mutation-testing-elements/test/unit/helpers/CustomElementFixture.ts [hunk 1]
```diff
@@ -47,4 +47,17 @@ export class CustomElementFixture<TCustomElement extends LitElement> {
   public dispose() {
     return this.element.remove();
   }
+
+  public async catchEvent<T extends Event = Event>(eventType: string, act: () => Promise<void>) {
```
Comment (nicojs) [inline packages/mutation-testing-elements/test/unit/helpers/CustomElementFixture.ts]: Love this little helper method 😍

File: packages/mutation-testing-report-schema/src/api/MutantResult.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { Location } from './Location';
 export interface MutantResult {
   id: string;
   mutatorName: string;
-  replacement: string;
+  replacement?: string;
```
Comment (nicojs) [inline packages/mutation-testing-report-schema/src/api/MutantResult.ts]: I'm still missing the changes in the schema and the unit tests to verify the schema.

**Comment by hcoles** [comment]
@nicojs yes, a dialog makes sense.

**Comment by nicojs** [comment]
@Wmaarts are you still on the ball for this one? ⚽️

**Comment by Wmaarts** [comment]
@nicojs I'll have a look at this thursday. 😁

File: packages/mutation-testing-elements/test/unit/helpers/CustomElementFixture.ts (hunk 1 above)
Comment (Wmaarts) [reply packages/mutation-testing-elements/test/unit/helpers/CustomElementFixture.ts]: 😁✨

File: packages/mutation-testing-elements/src/components/mutation-test-report-file/mutation-test-report-file.ts [hunk 3]
```diff
@@ -61,13 +64,24 @@ export class MutationTestReportFileComponent extends LitElement {
       const selectedMutant = (event as CustomEvent<MutationTestReportMutantComponent>).detail;
       this.forEachMutantComponent(mutant => mutant !== selectedMutant && (mutant.showPopup = false));
     });
+    this.addEventListener('show-more-click', (event: Event) => {
```
[the remark this answers is not in the document]
Comment (Wmaarts) [reply packages/mutation-testing-elements/src/components/mutation-test-report-file/mutation-test-report-file.ts]: Good idea, I'll implement it, however that makes it inconsistent with other eventlisteners defined here. :p

**Comment by Wmaarts** [comment]
@nicojs I made changes based on your feedback (good suggestions btw ^^). I think it's ready for review now.

**Comment by nicojs** [comment]
Thanks for the hard work! Will merge when the build is green 🍏💚📗🥗

**Pull request merged**